cli/src/test/java/com/devonfw/tools/ide/repo/CustomToolTest.java

```diff
@@ -64,4 +71,57 @@ public void testSpecific() {
     assertThat(tool.getChecksum()).isEqualTo(checksum);
   }
 
+  @Test
+  public void testReadCustomToolsFromJson() {
+    // arrange
+    AbstractIdeTestContext context = new IdeSlf4jContext(Path.of(""));
+    SystemInfo systemInfo = SystemInfoMock.of("linux");
+    context.setSystemInfo(systemInfo);
+    Path testPath = Path.of("src/test/resources/customtools");
+    // act
+    CustomToolsJson customToolsJson = CustomToolRepositoryImpl.readCustomToolsFromJson(context, testPath.resolve("custom-tools.json"));
+    // assert
+    assertThat(customToolsJson.url()).isEqualTo("https://some-file-server.company.com/projects/my-project");
+    assertThat(customToolsJson.tools().get(0).getUrl()).isEqualTo(
+        "https://some-file-server.company.com/projects/my-project/jboss-eap/7.1.4.GA/jboss-eap-7.1.4.GA.tgz");
+    assertThat(customToolsJson.tools().get(1).getUrl()).isEqualTo(
+        "https://some-file-server.company.com/projects/my-project2/firefox/70.0.1/firefox-70.0.1-linux-x64.tgz");
+  }
+
+  @Test
+  public void testReadCustomToolsFromLegacyConfig() {
+    // arrange
+    IdeContext context = new IdeTestContext();
+    String legacyProperties = "DEVON_IDE_CUSTOM_TOOLS=(jboss-eap:7.1.4.GA:all:https://host.tld/projects/my-project firefox:70.0.1:all:https://host.tld/projects/my-project2)";
```

Comment:
I already left review comments to Kian and Leon that the context of that method is kind of odd and also we need to reuse our existing infrastructure to parse `ide.properties` and `devon.properties` files.
Then we can just get the value of a variable and only need to pass that value to this method instead of fiddling the value from the variable inside a class that is not responsible for that and code gets more redundant.
BTW: We also have existing code to parse variable array syntax into a regular Java `List`. This is currently in `VariableDefinitionStringList` e.g. used by `IdeVariables.IDE_TOOLS` to parse things like `IDE_TOOLS=(java mvn node npm)` (see https://github.com/devonfw/IDEasy/blob/main/documentation/variables.adoc).
We could even extract that parsing code to `VariableLine` for easier reuse here.

From the test perspective I would then only pass the value and not the entire variable declaration line.